src/Module/Settings/Uexport.php

```diff
@@ -0,0 +1,207 @@
+{
+	/**
+	 * Handle the request to export data.
+	 * At the moment one can export two different data set
+	 * 1. The profile data that can be used by uimport to resettle
+	 *    to a different Friendica instance
+	 * 2. The entire data-set, profile plus postings
+	 *
+	 * If there is an action required through the URL / path, react
+	 * accordingly and export the requested data.
+	 **/
+	public static function content()
+	{
+		parent::content();
+		$args = self::getClass(Arguments::class);
+		if ($args->getArgc() == 3) {
+			// @TODO Replace with router-provided arguments
+			$action = $args->get(2);
+			header("Content-type: application/json");
+			header('Content-Disposition: attachment; filename="' . $a->user['nickname'] . '.' . $action . '"');
+			switch ($action) {
+				case "backup":
+					SELF::uexport_all($a);
+					exit();
+					break;
+				case "account":
+					SELF::uexport_account($a);
+					exit();
+					break;
+				default:
+					exit();
+			}
+		}
+
+		/**
+		 * options shown on "Export personal data" page
+		 * list of array( 'link url', 'link text', 'help text' )
+		 */
+		$options = [
+			['settings/uexport/account', L10n::t('Export account'), L10n::t('Export your account info and contacts. Use this to make a backup of your account and/or to move it to another server.')],
+			['settings/uexport/backup', L10n::t('Export all'), L10n::t("Export your accout info, contacts and all your items as json. Could be a very big file, and could take a lot of time. Use this to make a full backup of your account \x28photos are not exported\x29")],
+		];
+		Hook::callAll('uexport_options', $options);
+
+		$tpl = Renderer::getMarkupTemplate("uexport.tpl");
+		return Renderer::replaceMacros($tpl, [
+			'$title' => L10n::t('Export personal data'),
+			'$options' => $options
+		]);
+	}
+	private function uexport_multirow($query) {
+		global $dbStructure;
+
+		preg_match("/\s+from\s+`?([a-z\d_]+)`?/i", $query, $match);
+		$table = $match[1];
+
+		$result = [];
+		$r = q($query);
+		if (DBA::isResult($r)) {
+			foreach ($r as $rr) {
+				$p = [];
+				foreach ($rr as $k => $v) {
+					switch ($dbStructure[$table]['fields'][$k]['type']) {
+						case 'datetime':
+							$p[$k] = $v ?? DBA::NULL_DATETIME;
+							break;
+						default:
+							$p[$k] = $v;
+							break;
+					}
+				}
+				$result[] = $p;
+			}
+		}
+		return $result;
+	}
+
+	private function uexport_row($query) {
+		global $dbStructure;
+
+		preg_match("/\s+from\s+`?([a-z\d_]+)`?/i", $query, $match);
+		$table = $match[1];
+
+		$result = [];
+		$r = q($query);
+		if (DBA::isResult($r)) {
+
+			foreach ($r as $rr) {
+				foreach ($rr as $k => $v) {
+					switch ($dbStructure[$table]['fields'][$k]['type']) {
+						case 'datetime':
+							$result[$k] = $v ?? DBA::NULL_DATETIME;
+							break;
+						default:
+							$result[$k] = $v;
+							break;
+					}
+				}
+			}
+		}
+		return $result;
+	}
+
+	private function uexport_account($a) {
+
+		$user = SELF::uexport_row(
```

Comment:
Goes for all occurences of this method: lowercase to `self::`

Reply:
Yes, and I wonder why you repeat your comment. The first comment was enough to point it out.

Reply:
To have GitHub keep track of all the occurrences. All the comments related to changed lines will be automatically folded while unmodified line comments will stay open in the web view.